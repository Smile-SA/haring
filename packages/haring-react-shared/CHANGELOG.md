# @smile/haring-react-shared

## 0.15.0

### Minor Changes

- [#194](https://github.com/Smile-SA/haring/pull/194) [`97a610a`](https://github.com/Smile-SA/haring/commit/97a610aa02b86423b723ef8ae5d45e8ecf346f92) Thanks [@vapersmile](https://github.com/vapersmile)! - Upgrade mantine version to 7.11.x

## 0.14.0

## 0.13.0

### Minor Changes

- [#178](https://github.com/Smile-SA/haring/pull/178) [`ce8f39a2c52590b2c91950450a84222434913743`](https://github.com/Smile-SA/haring/commit/ce8f39a2c52590b2c91950450a84222434913743) Thanks [@tonai](https://github.com/tonai)! - Rename packages

## 0.12.0

## 0.11.0

## 0.10.0

### Minor Changes

- [#149](https://github.com/Smile-SA/haring/pull/149) [`2706a09`](https://github.com/Smile-SA/haring/commit/2706a097fd86b072cf21d12fe2a97427883671cf) Thanks [@vapersmile](https://github.com/vapersmile)! - Add Calendar style on globlaStyle and add a preview

- [#150](https://github.com/Smile-SA/haring/pull/150) [`34fabfb`](https://github.com/Smile-SA/haring/commit/34fabfbd1a2c048c39adc567548a3ee7e85074ee) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Fixed missing global styles for focus-visible color

## 0.9.0

### Minor Changes

- [`d430e2c`](https://github.com/Smile-SA/haring/commit/d430e2cbcd26871c4db0c53f92a4e4baa6af839c) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Migrated from Mantine V6 to V7

## 0.8.1

## 0.8.0

### Minor Changes

- [#108](https://github.com/Smile-SA/haring/pull/108) [`320634d`](https://github.com/Smile-SA/haring/commit/320634d7ec2048a328cfda1dbcff5d1b70a32563) Thanks [@vapersmile](https://github.com/vapersmile)! - Add custom default style for carousel mantine composent and add 2-Mantine/carousel page

## 0.7.0

### Minor Changes

- [#87](https://github.com/Smile-SA/haring/pull/87) [`dbc06ca`](https://github.com/Smile-SA/haring/commit/dbc06ca55961b69663ab7fdc02609c6525ae361d) Thanks [@tonai](https://github.com/tonai)! - Add theme provider and update barrel file for better next.js compatibility

- [#95](https://github.com/Smile-SA/haring/pull/95) [`5696400`](https://github.com/Smile-SA/haring/commit/5696400e6f703da52db5f7199b50f8251fa76726) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `FilterList` component, renamed `FiltersBar` in `SidebarFilters`
  and `FiltersCheckboxList` in `SearchableCheckboxList`,
  refactored `SearchableCheckboxList`, added common `IFilter` type in shared
  package, added topContent prop to `SwitchableView`, fixed `BrowsingPage` with
  name changes.

## 0.6.0

### Minor Changes

- [#74](https://github.com/Smile-SA/haring/pull/74) [`0ad5377`](https://github.com/Smile-SA/haring/commit/0ad5377535e7c19941da301e8a2ee7298ab70f91) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Refactored actions in `ThumbnailGrid` and `TableGridView` components, moved `typeGuard` helper to shared package, added `typeGuardInterface` function, added tests

## 0.5.0

### Minor Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add dynamic componentProps property in actions

- [#65](https://github.com/Smile-SA/haring/pull/65) [`bd9cfe4`](https://github.com/Smile-SA/haring/commit/bd9cfe42d0b22f5f7f5e7b0de30fdfb22ad3e1c8) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Reworked action/confirm types and moved then in `react-front-kit-shared` then refactored types in `Table`, `Thumbnail` and `ConfirmModal` , Added `ThumbnailGrid` and `TableGridView` components,

- [#68](https://github.com/Smile-SA/haring/pull/68) [`c8b5a39`](https://github.com/Smile-SA/haring/commit/c8b5a3978c8ce7133da0b498d9f0b326f07eb737) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `storybook-pages` package containing example pages, renamed `TestPage`
  into `BrowsingPage` and completed it with integration of `TableGridView`, fixed
  various component styles and props related to page.

### Patch Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add server barrel files

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/haring/pull/53) [`01ee3ea`](https://github.com/Smile-SA/haring/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
