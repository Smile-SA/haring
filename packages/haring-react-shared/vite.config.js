import { resolve } from 'path';

import react from '@vitejs/plugin-react-swc';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.tsx'),
      fileName: '[name]',
      formats: ['cjs', 'es'],
    },
    rollupOptions: {
      external: [
        '@mantine/carousel',
        '@mantine/core',
        '@mantine/dates',
        '@mantine/dropzone',
        '@mantine/hooks',
        '@phosphor-icons/react',
        '@smile/haring-react',
        '@smile/haring-react-dates',
        '@smile/haring-react-dropzone',
        '@smile/haring-react-shared',
        '@smile/haring-react-table',
        '@storybook/addon-actions',
        '@storybook/preview-api',
        '@tabler/icons-react',
        '@testing-library/react',
        'dayjs',
        'embla-carousel-react',
        'mantine-react-table',
        'react',
        'react/jsx-runtime',
      ],
      input: {
        index: resolve(__dirname, 'src/index.tsx'),
        server: resolve(__dirname, 'src/server.tsx'),
        'storybook-utils': resolve(__dirname, 'src/storybook-utils/index.ts'),
        'test-utils': resolve(__dirname, 'src/test-utils/index.ts'),
      },
      output: {
        banner: (chunkInfo) => {
          if (
            ['src/index.tsx'].find((modulePath) =>
              chunkInfo.facadeModuleId?.endsWith(modulePath),
            )
          ) {
            return `"use client"`;
          }
          return '';
        },
        preserveModules: false,
      },
    },
  },
  plugins: [
    react(),
    dts({
      insertTypesEntry: true,
      tsconfigPath: './tsconfig.build.json',
    }),
  ],
});
