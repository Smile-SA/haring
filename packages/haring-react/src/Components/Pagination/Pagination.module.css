.container {
  justify-content: space-between;

  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    flex-direction: column;
    row-gap: 16px;
  }
}

.pagination {
  display: flex;
  justify-content: center;
}

.transparent {
  background-color: transparent;
  border-color: transparent;

  &[data-active='true'] {
    background-color: var(--mantine-primary-color-filled);
  }
}
