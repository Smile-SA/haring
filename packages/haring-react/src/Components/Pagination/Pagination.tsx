'use client';
import type { FlexProps, PaginationProps, SelectProps } from '@mantine/core';
import type { IOptions } from '@smile/haring-react-shared';
import type { ReactElement } from 'react';

import { Flex, Pagination as MantinePagination, Select } from '@mantine/core';

import classes from './Pagination.module.css';

export interface IPaginationProps extends FlexProps {
  isTransparent?: boolean;
  itemsPerPage: number;
  itemsPerPageAriaLabel?: string;
  itemsPerPageOptions?: IOptions<number>;
  onItemsPerPageChange?: (value: number) => void;
  onPageChange?: (value: number) => void;
  page: number;
  paginationProps?: PaginationProps;
  selectProps?: SelectProps;
  totalPages: number;
}

/** Additional props will be forwarded to the [Mantine Flex component](https://mantine.dev/core/flex) */
export function Pagination(props: IPaginationProps): ReactElement {
  const {
    isTransparent = false,
    itemsPerPage,
    itemsPerPageAriaLabel,
    itemsPerPageOptions = [],
    onItemsPerPageChange,
    onPageChange,
    page,
    paginationProps,
    selectProps,
    totalPages,
    ...flexProps
  } = props;

  /* methods */
  function handleChangeItemsPerPage(value: string | null): void {
    if (value) {
      onItemsPerPageChange?.(Number(value));
    }
  }

  function handleChangePage(value: number): void {
    onPageChange?.(value);
  }

  return (
    <Flex className={classes.container} data-testid="pagination" {...flexProps}>
      {itemsPerPageOptions.length > 0 && itemsPerPageAriaLabel ? (
        <Select
          aria-label={itemsPerPageAriaLabel}
          data={itemsPerPageOptions.map((option) => {
            return {
              label: option.label ?? option.value.toString(),
              value: option.value.toString(),
            };
          })}
          data-testid="pagination-itemsPerPage"
          onChange={handleChangeItemsPerPage}
          size="xs"
          value={itemsPerPage.toString()}
          {...selectProps}
        />
      ) : null}
      <MantinePagination
        autoContrast
        className={classes.pagination}
        classNames={{ control: isTransparent ? classes.transparent : '' }}
        data-testid="pagination-page"
        onChange={handleChangePage}
        radius="sm"
        size="sm"
        total={totalPages}
        value={page}
        withControls={false}
        {...paginationProps}
      />
    </Flex>
  );
}
