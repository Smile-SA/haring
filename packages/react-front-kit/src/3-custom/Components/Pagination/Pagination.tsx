'use client';
import type { PaginationProps } from '@mantine/core';
import type { FlexProps } from '@mantine/core/lib/Flex/Flex';
import type { SelectProps } from '@mantine/core/lib/Select/Select';
import type { ReactElement } from 'react';

import {
  Flex,
  Pagination as MantinePagination,
  Select,
  createStyles,
} from '@mantine/core';

const useStyles = createStyles(() => ({
  container: {
    justifyContent: 'space-between',
  },
}));

export interface IRowsPerPageOption {
  label?: string;
  value: number;
}

interface IPaginationProps extends FlexProps {
  onPageChange?: (value: number) => void;
  onRowsPerPageChange?: (value: number) => void;
  page: number;
  paginationProps?: PaginationProps;
  rowsPerPage: number;
  rowsPerPageLabel?: string;
  rowsPerPageOptions?: IRowsPerPageOption[];
  selectProps?: SelectProps;
  totalPages: number;
}

export function Pagination(props: IPaginationProps): ReactElement {
  const {
    onPageChange,
    onRowsPerPageChange,
    page,
    paginationProps,
    rowsPerPage,
    rowsPerPageLabel,
    rowsPerPageOptions = [],
    selectProps,
    totalPages,
    ...flexProps
  } = props;
  const { classes } = useStyles();

  /* methods */
  function handleChangeRowsPerPage(value: string): void {
    if (value) {
      onRowsPerPageChange?.(Number(value));
    }
  }

  function handleChangePage(value: number): void {
    onPageChange?.(value);
  }

  return (
    <Flex className={classes.container} data-testid="pagination" {...flexProps}>
      {rowsPerPageOptions.length > 0 && rowsPerPageLabel ? (
        <Select
          aria-label={rowsPerPageLabel}
          data={rowsPerPageOptions.map((option) => {
            return {
              label: option.label ?? option.value.toString(),
              value: option.value.toString(),
            };
          })}
          data-testid="pagination-rowsPerPage"
          onChange={handleChangeRowsPerPage}
          size="xs"
          value={rowsPerPage.toString()}
          {...selectProps}
        />
      ) : null}
      <MantinePagination
        data-testid="pagination-page"
        onChange={handleChangePage}
        radius="sm"
        size="sm"
        total={totalPages}
        value={page}
        withControls={false}
        {...paginationProps}
      />
    </Flex>
  );
}
