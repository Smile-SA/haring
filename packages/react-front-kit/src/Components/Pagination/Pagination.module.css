.container {
  justify-content: space-between;

  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    flex-direction: column;
    row-gap: 16px;
  }
}

.pagination {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    display: flex;
    justify-content: center;
  }
}
