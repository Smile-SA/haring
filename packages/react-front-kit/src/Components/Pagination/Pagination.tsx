'use client';
import type { PaginationProps } from '@mantine/core';
import type { FlexProps } from '@mantine/core/lib/Flex/Flex';
import type { SelectProps } from '@mantine/core/lib/Select/Select';
import type { IOptions } from '@smile/react-front-kit-shared/src/types/options';
import type { ReactElement } from 'react';

import {
  Flex,
  Pagination as MantinePagination,
  Select,
  createStyles,
} from '@mantine/core';

const useStyles = createStyles(() => ({
  container: {
    justifyContent: 'space-between',
  },
}));

export interface IPaginationProps extends FlexProps {
  itemsPerPage: number;
  itemsPerPageAriaLabel?: string;
  itemsPerPageOptions?: IOptions<number>;
  onItemsPerPageChange?: (value: number) => void;
  onPageChange?: (value: number) => void;
  page: number;
  paginationProps?: PaginationProps;
  selectProps?: SelectProps;
  totalPages: number;
}

/** Additional props will be forwarded to the [Mantine Flex component](https://mantine.dev/core/flex) */
export function Pagination(props: IPaginationProps): ReactElement {
  const {
    itemsPerPage,
    itemsPerPageAriaLabel,
    itemsPerPageOptions = [],
    onItemsPerPageChange,
    onPageChange,
    page,
    paginationProps,
    selectProps,
    totalPages,
    ...flexProps
  } = props;
  const { classes } = useStyles();

  /* methods */
  function handleChangeItemsPerPage(value: string): void {
    if (value) {
      onItemsPerPageChange?.(Number(value));
    }
  }

  function handleChangePage(value: number): void {
    onPageChange?.(value);
  }

  return (
    <Flex className={classes.container} data-testid="pagination" {...flexProps}>
      {itemsPerPageOptions.length > 0 && itemsPerPageAriaLabel ? (
        <Select
          aria-label={itemsPerPageAriaLabel}
          data={itemsPerPageOptions.map((option) => {
            return {
              label: option.label ?? option.value.toString(),
              value: option.value.toString(),
            };
          })}
          data-testid="pagination-itemsPerPage"
          onChange={handleChangeItemsPerPage}
          size="xs"
          value={itemsPerPage.toString()}
          {...selectProps}
        />
      ) : null}
      <MantinePagination
        data-testid="pagination-page"
        onChange={handleChangePage}
        radius="sm"
        size="sm"
        total={totalPages}
        value={page}
        withControls={false}
        {...paginationProps}
      />
    </Flex>
  );
}
