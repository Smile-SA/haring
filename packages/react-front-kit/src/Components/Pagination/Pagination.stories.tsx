import type { Meta, StoryObj } from '@storybook/react';

import { useStorybookArgsConnect } from '@smile/react-front-kit-shared/src/storybook-utils';

import { Pagination as Cmp } from './Pagination';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onPageChange: 'page',
        onRowsPerPageChange: 'rowsPerPage',
      });
      return <Story args={{ ...args }} />;
    },
  ],
  tags: ['autodocs'],
  title: '3-custom/Components/Pagination',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const Pagination: IStory = {
  args: {
    page: 2,
    rowsPerPage: 15,
    rowsPerPageLabel: 'Number of results per page',
    rowsPerPageOptions: [
      { label: 'Display 1 result', value: 1 },
      { label: 'Display 5 results', value: 5 },
      { label: 'Display 10 results', value: 10 },
      { label: 'Display 15 results', value: 15 },
    ],
    totalPages: 10,
  },
};
