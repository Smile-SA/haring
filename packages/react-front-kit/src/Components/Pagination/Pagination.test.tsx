import { renderWithProviders } from '@smile/haring-react-shared/test-utils';
import { within } from '@testing-library/react';

import { Pagination } from './Pagination';

describe('Pagination', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });

  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <Pagination
        itemsPerPage={15}
        itemsPerPageAriaLabel="Number of results per page"
        itemsPerPageOptions={[
          { label: 'Display 1 result', value: 1 },
          { label: 'Display 5 results', value: 5 },
          { label: 'Display 10 results', value: 10 },
          { label: 'Display 15 results', value: 15 },
        ]}
        page={2}
        totalPages={10}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with minimal props', () => {
    const { container } = renderWithProviders(
      <Pagination itemsPerPage={15} page={2} totalPages={10} />,
    );
    const canvas = within(container);
    expect(canvas.queryByTestId('pagination')).toBeVisible();
    expect(canvas.queryByTestId('pagination-itemsPerPage')).toBeNull();
    expect(canvas.queryByTestId('pagination-page')).toBeVisible();
  });

  it('renders with full props', () => {
    const { container } = renderWithProviders(
      <Pagination
        itemsPerPage={15}
        itemsPerPageAriaLabel="Number of results per page"
        itemsPerPageOptions={[
          { value: 1 },
          { value: 5 },
          { value: 10 },
          { value: 15 },
        ]}
        page={2}
        totalPages={10}
      />,
    );
    const canvas = within(container);
    expect(canvas.queryByTestId('pagination')).toBeVisible();
    expect(canvas.queryByTestId('pagination-itemsPerPage')).toBeVisible();
    expect(canvas.queryByTestId('pagination-page')).toBeVisible();
  });
});
