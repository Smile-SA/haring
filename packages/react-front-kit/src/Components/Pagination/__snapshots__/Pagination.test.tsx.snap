// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<div>
  <div
    class="mantine-brykyr"
    data-testid="pagination"
  >
    <div
      class="mantine-InputWrapper-root mantine-Select-root mantine-q6oiko"
    >
      <div
        aria-controls="mantine-f4bipx4bi"
        aria-expanded="false"
        aria-haspopup="listbox"
        class=""
        role="combobox"
        tabindex="-1"
      >
        <input
          type="hidden"
          value="15"
        />
        <div
          class="mantine-Input-wrapper mantine-Select-wrapper mantine-1v7s5f8"
        >
          <input
            aria-autocomplete="list"
            aria-invalid="false"
            aria-label="Number of results per page"
            autocomplete="off"
            class="mantine-Input-input mantine-Select-input mantine-1bcy86t"
            data-mantine-stop-propagation="false"
            data-testid="pagination-itemsPerPage"
            id="mantine-f4bipx4bi"
            readonly=""
            type="search"
            value="Display 15 results"
          />
          <div
            class="mantine-1wnb5rr mantine-Input-rightSection mantine-Select-rightSection"
          >
            <svg
              data-chevron="true"
              fill="none"
              style="color: rgb(134, 142, 150); width: 0.875rem; height: 0.875rem;"
              viewBox="0 0 15 15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mantine-Group-root mantine-uxw5g5"
      data-testid="pagination-page"
    >
      <button
        class="mantine-UnstyledButton-root mantine-Pagination-control mantine-v1qgro"
        type="button"
      >
        1
      </button>
      <button
        aria-current="page"
        class="mantine-UnstyledButton-root mantine-Pagination-control mantine-v1qgro"
        data-active="true"
        type="button"
      >
        2
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Pagination-control mantine-v1qgro"
        type="button"
      >
        3
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Pagination-control mantine-v1qgro"
        type="button"
      >
        4
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Pagination-control mantine-v1qgro"
        type="button"
      >
        5
      </button>
      <div
        class="mantine-Pagination-dots mantine-90n090"
      >
        <svg
          style="width: calc(1.625rem / 2); height: calc(1.625rem / 2);"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 8c0-.733.6-1.333 1.333-1.333.734 0 1.334.6 1.334 1.333s-.6 1.333-1.334 1.333C2.6 9.333 2 8.733 2 8zm9.333 0c0-.733.6-1.333 1.334-1.333C13.4 6.667 14 7.267 14 8s-.6 1.333-1.333 1.333c-.734 0-1.334-.6-1.334-1.333zM6.667 8c0-.733.6-1.333 1.333-1.333s1.333.6 1.333 1.333S8.733 9.333 8 9.333 6.667 8.733 6.667 8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <button
        class="mantine-UnstyledButton-root mantine-Pagination-control mantine-v1qgro"
        type="button"
      >
        10
      </button>
    </div>
  </div>
</div>
`;
